go/oasis-node/cmd/debug/byzantine/tendermint.go

```diff
@@ -23,123 +21,51 @@ import (
-
-// GetBaseEpoch implements epochtime Backend.
-func (t *fakeTimeBackend) GetBaseEpoch(ctx context.Context) (epochtime.EpochTime, error) {
-	// XXX: This will need to honor the base epoch if this is ever used in
-	// conjunction with the real timekeeping backend, and a dump/restore.
-	return 0, nil
-}
-
-// GetEpoch implements epochtime Backend.
-func (t *fakeTimeBackend) GetEpoch(ctx context.Context, height int64) (epochtime.EpochTime, error) {
-	if height == 0 {
-		panic("0 height not supported")
-	}
-
-	if t.useMockEpochTime {
-		// Query the epochtime_mock Tendermint application.
-		q, err := t.service.epochtimeMockQuery.QueryAt(ctx, height)
-		if err != nil {
-			return epochtime.EpochInvalid, errors.Wrap(err, "epochtime: epoch query failed")
-		}
-
-		epoch, _, err := q.Epoch(ctx)
-		if err != nil {
-			return epochtime.EpochInvalid, errors.Wrap(err, "epochtime: epoch query failed")
-		}
-		return epoch, nil
-	}
-
-	// Use the the epoch interval that we have in E2E tests.
-	// We could make this more flexible with command line flags in future work.
-	return epochtime.EpochTime(height / 30), nil
-}
-
-// GetEpochBlock implements epochtime Backend.
-func (*fakeTimeBackend) GetEpochBlock(ctx context.Context, epoch epochtime.EpochTime) (int64, error) {
-	panic("GetEpochBlock not supported")
-}
-
-// WatchEpochs implements epochtime Backend.
-func (*fakeTimeBackend) WatchEpochs() (<-chan epochtime.EpochTime, *pubsub.Subscription) {
-	panic("WatchEpochs not supported")
-}
-
-func (*fakeTimeBackend) ToGenesis(ctx context.Context, height int64) (*epochtime.Genesis, error) {
-	panic("ToGenesis not supported")
-}
-
 type honestTendermint struct {
 	service service.TendermintService
 
-	schedulerQuery     *schedulerapp.QueryFactory
-	roothashQuery      *roothashapp.QueryFactory
-	registryQuery      *registryapp.QueryFactory
-	epochtimeMockQuery *epochtimemockapp.QueryFactory
+	schedulerQuery *schedulerapp.QueryFactory
+	roothashQuery  *roothashapp.QueryFactory
+	registryQuery  *registryapp.QueryFactory
 }
 
 func newHonestTendermint() *honestTendermint {
 	return &honestTendermint{}
 }
 
-func (ht *honestTendermint) start(id *identity.Identity, dataDir string, useMockEpochTime bool) error {
+func (ht *honestTendermint) start(id *identity.Identity, dataDir string) error {
 	if ht.service != nil {
 		return errors.New("honest Tendermint service already started")
 	}
 
-	genesis, err := genesis.New()
+	genesis, err := genesis.NewFileProvider()
 	if err != nil {
-		return errors.Wrap(err, "genesis New")
+		return errors.Wrap(err, "genesis NewFileProvider")
 	}
 	ht.service, err = tendermint.New(context.Background(), dataDir, id, genesis)
 	if err != nil {
 		return errors.Wrap(err, "tendermint New")
 	}
 
-	if err = ht.service.ForceInitialize(); err != nil {
-		return errors.Wrap(err, "honest Tendermint service ForceInitialize")
-	}
-
 	// Register honest mux apps.
 	// This isn't very flexible. It's configured to match what we use in end-to-end tests.
```

Comment:
Should this comment be updated now that options come from genesis?

Reply:
yeah can probably just remove the whole comment? (keeping the `// Register honest mux apps.`)

Reply:
:+1: